promotion-guide.adoc

```diff
@@ -0,0 +1,137 @@
+
+The environment promotion feature of the MRO automates moving a certain version of the software from one environment to the next. Developers only have to tell the MRO if a new version should be built (in DEV) and packaged as an installable "release bundle", or if an existing "release bundle" should be promoted to either the QA or the production environment.
+
+The environment promotion feature is part of the regular MRO pipeline. Therefore, the promotion is executed from various Jenkins stages. It is not possible to change the process itself, but you can customize how the prmotion happens exactly for each of your software components.
+
+## Source Code Organisation
+
+The components of your software are defined in the `repositories` section of the `metadata.yml` file in the release manager repository. In order for the MRO to know which state of each component should be promoted, it needs to have some knowledge about how version control in your repositories is organised. Everything depends on a user-supplied input parameter named `version` to the pipeline. Other input parameters do not have any impact on source code lookup.
+
+- When no `version` is given, the MRO will default to `WIP` (work in progress). In this scenario, source code for each repository is taken from the configured branch in the `metadata.yml` file (defaulting to `master` if no branch is specified there).
+- When a `version` is given, source code will be taken from a branch `release/$VERSION` in each repository. When this branch does not exist yet, it will be created (based on the configured branch in `metadata.yml`) by the pipeline. Subsequent runs with the same `version` input will take the source code from the created release branch - changes to the configured branch will have no effect on this version! This is by design: it allows some developers to work on new features on the mainline branch (typically `master`) while others polish the release branch. To this end, the MRO allows to enable separate development environments per version to isolate changes in OpenShift resources (see section "Environments" further down).
+- The MRO applies the same branching rules to its own repository - it will create a release branch per version. There is one small caveat here: Jenkins downloads the `Jenkinsfile` of a pipeline from the branch which is configured for this pipeline. That means that for a pipeline setup against `master`, Jenkins will always execute the latest `Jenkinsfile` from `master`, even when you pass an  explicit `version` to the pipeline. The MRO will read e.g. the `metadata.yml` file from the matching release branch, but the `Jenkinsfile` itself will be from `master`. Usually, this should not be an issue as you should not make changes to the `Jenkinsfile` anyway.
```

Comment:
@michaelsauter  When you say:

> Jenkins downloads the `Jenkinsfile` of a pipeline from the branch which is configured for this pipeline. That means that for a pipeline setup against `master` ...

MRO gets the `Jenkinsfile` of `master` for all the repositories, or just for the release manager one?